gcloud/streaming/transfer.py

```diff
@@ -1,37 +1,54 @@
 # pylint: skip-file
-"""Upload and download support for apitools."""
-from __future__ import print_function
+"""Upload and download support for apitools.
+
+:mod:`gcloud.storage.blob` uses:
+
+- :class:`Download`
+- :class:`Upload`
+- :const:`RESUMABLE_UPLOAD`
+
+:mod:`gcloud.storage.test_blob` patches:
+
+- :const:`RESUMABLE_UPLOAD_THRESHOLD`
+"""
 
 import email.generator as email_generator
 import email.mime.multipart as mime_multipart
 import email.mime.nonmultipart as mime_nonmultipart
-import io
-import json
 import mimetypes
 import os
-import threading
 
 import six
 from six.moves import http_client
 
-from gcloud.streaming import buffered_stream
-from gcloud.streaming import exceptions
-from gcloud.streaming import http_wrapper
-from gcloud.streaming import stream_slice
-from gcloud.streaming import util
+from gcloud.streaming.buffered_stream import BufferedStream
+from gcloud.streaming.exceptions import CommunicationError
+from gcloud.streaming.exceptions import ConfigurationValueError
+from gcloud.streaming.exceptions import HttpError
+from gcloud.streaming.exceptions import InvalidDataError
+from gcloud.streaming.exceptions import InvalidUserInputError
+from gcloud.streaming.exceptions import NotFoundError
+from gcloud.streaming.exceptions import TransferInvalidError
+from gcloud.streaming.exceptions import TransferRetryError
+from gcloud.streaming.exceptions import UserError
+from gcloud.streaming.http_wrapper import get_http
+from gcloud.streaming.http_wrapper import handle_http_exceptions
+from gcloud.streaming.http_wrapper import make_api_request
+from gcloud.streaming.http_wrapper import Request
+from gcloud.streaming.http_wrapper import RESUME_INCOMPLETE
+from gcloud.streaming.stream_slice import StreamSlice
+from gcloud.streaming.util import acceptable_mime_type
+from gcloud.streaming.util import type_check
```

Comment:
I still prefer to name the imports explicitly.  Some of these imports could actually go away:

- The funky exception hierarchy could pretty much go away entirely:  we don't catch / handle those errors anywhere in our code.
- Likely `get_http` (the factory indirection) and `type_check` (the Anti-Duck-Typing thing you note below) could go.